var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py

```diff
@@ -49,32 +52,39 @@ class PyFenicsDolfinx(PythonPackage):
     depends_on("fenics-dolfinx@0.1.0", when="@0.1.0")
 
     depends_on("fenics-basix@main", type=("build", "link"), when="@main")
+    depends_on("fenics-basix@0.6.0:0.6", type=("build", "link"), when="@0.6.0:0.6")
     depends_on("fenics-basix@0.5.1:0.5", type=("build", "link"), when="@0.5.0:0.5")
     depends_on("fenics-basix@0.4.2", type=("build", "link"), when="@0.4.1")
     depends_on("fenics-basix@0.3.0", type=("build", "link"), when="@0.3.0")
     depends_on("fenics-basix@0.2.0", type=("build", "link"), when="@0.2.0")
     depends_on("fenics-basix@0.1.0", type=("build", "link"), when="@0.1.0")
 
     depends_on("py-fenics-ffcx@main", type="run", when="@main")
+    depends_on("py-fenics-ffcx@0.6", type="run", when="@0.6")
     depends_on("py-fenics-ffcx@0.5.0.post0", type="run", when="@0.5.0:0.5")
     depends_on("py-fenics-ffcx@0.4.2", type="run", when="@0.4.1")
     depends_on("py-fenics-ffcx@0.3.0", type="run", when="@0.3.0")
     depends_on("py-fenics-ffcx@0.2.0", type="run", when="@0.2.0")
     depends_on("py-fenics-ffcx@0.1.0", type="run", when="@0.1.0")
 
     depends_on("py-fenics-ufl@main", type="run", when="@main")
-    depends_on("py-fenics-ufl@2022.2.0", type="run", when="@0.5.0:0.5")
-    depends_on("py-fenics-ufl@2022.1.0", type="run", when="@0.4.1")
-    depends_on("py-fenics-ufl@2021.1.0", type="run", when="@0.1:0.3.99")
+    depends_on("py-fenics-ufl@2023.1", type="run", when="@0.6")
+    depends_on("py-fenics-ufl@2022.2.0", type=("build", "run"), when="@0.5.0:0.5")
+    depends_on("py-fenics-ufl@2022.1.0", type=("build", "run"), when="@0.4.1")
+    depends_on("py-fenics-ufl@2021.1.0", type=("build", "run"), when="@0.1:0.3.99")
 
     depends_on("py-numpy@1.21:", type=("build", "run"), when="@0.5.0:")
-    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-numpy@1.15:", type=("build", "run"))
 
     depends_on("py-mpi4py", type=("build", "run"))
     depends_on("py-petsc4py", type=("build", "run"))
-    depends_on("py-pybind11@2.6.2:", type=("build", "run"))
+    depends_on("py-pybind11@2.7.0:", type=("build", "run"), when="@0.4:")
+    depends_on("py-pybind11@2.2.4:", type=("build", "run"))
     depends_on("xtensor@0.23.10:", type="build", when="@:0.5")
 
+    depends_on("py-setuptools@42:", type=("build", "run"), when="@0.4:")
+    depends_on("py-setuptools@40:", type=("build", "run"))
```

Comment:
Is the minimum setuptools version documented anywhere?

Reply:
It hasn't. Advice on this would be appreciated.

Reply:
If you know that this is the correct min version, then it's fine to include in the package. I would suggest updating the fenics source code to correctly document all dependencies someday.